source/internal.d.ts

```diff
@@ -357,6 +441,186 @@ IsPrimitive<Object>
+	? false
+	: true extends [IsEqual<A, true>, IsEqual<B, true>][number]
+		? true
+		: never;
+
+/**
+Returns a boolean for whether A is false.
+
+@example
+```
+Not<true>
+//=> false
+
+Not<false>
+//=> true
+```
+*/
+export type Not<A extends boolean> = A extends true
+	? false
+	: A extends false
+		? true
+		: never;
+
+/**
+Returns the maximum value from a tuple of integers.
+
+Note:
+- float number is not supported.
+
+@example
+```
+ArrayMax<[1, 2, 5, 3]>;
+//=> 5
+
+ArrayMax<[1, 2, 5, 3, 99, -1]>
+//=> 99
+```
+*/
+export type ArrayMax<A extends number[], Result extends number = NegativeInfinity> = number extends A[number]
+	? never :
+	A extends [infer F extends number, ...infer R extends number[]]
+		? Gt<F, Result> extends true
+			? ArrayMax<R, F>
+			: ArrayMax<R, Result>
+		: Result;
+
+/**
+Returns the minimal value from a tuple of integers.
+
+@example
+```
+ArrayMin<[1, 2, 5, 3]>
+//=> 1
+
+ArrayMin<[1, 2, 5, 3, -5]>
+//=> -5
+```
+*/
+export type ArrayMin<A extends number[], Result extends number = PositiveInfinity> = number extends A[number]
+	? never
+	: A extends [infer F extends number, ...infer R extends number[]]
+		? Lt<F, Result> extends true
+			? ArrayMin<R, F>
+			: ArrayMin<R, Result>
+		: Result;
+
+/**
+Returns the absolute value of a given value.
+
+@example
+```
+NumberAbsolute<-1>
+//=> 1
+
+NumberAbsolute<1>
+//=> 1
+
+NumberAbsolute<NegativeInfinity>
+//=> PositiveInfinity
+```
+*/
+export type NumberAbsolute<N extends number> = `${N}` extends `-${infer StringPositiveN}` ? StringToNumber<StringPositiveN> : N;
+
+/**
+Returns a boolean for whether A > B(A and B are both numeric string and have the same length).
+
+@example
+```
+SameLengthPositiveNumericStringGt<'50', '10'>
+//=> true
+
+SameLengthPositiveNumericStringGt<'10', '10'>
+//=> false
+```
+*/
+type SameLengthPositiveNumericStringGt<A extends string, B extends string> = A extends `${infer FirstA}${infer RestA}`
+	? B extends `${infer FirstB}${infer RestB}`
+		? FirstA extends FirstB
+			? SameLengthPositiveNumericStringGt<RestA, RestB>
+			: PositiveNumericCharGt<FirstA, FirstB>
+		: never
+	: false;
+
+type NumericString = '0123456789';
+
+/**
+Returns a boolean for whether A > B(A and B are both positive numeric string).
+
+@example
+```
+PositiveNumericStringGt<'500', '1'>
+//=> true
+
+PositiveNumericStringGt<'1', '1'>
+//=> false
+
+PositiveNumericStringGt<'1', '500'>
+//=> false
+```
+*/
+export type PositiveNumericStringGt<A extends string, B extends string> = A extends B
+	? false
+	: [BuildTuple<StringLength<A>, 0>, BuildTuple<StringLength<B>, 0>] extends infer R extends [readonly unknown[], readonly unknown[]]
+		? R[0] extends [...R[1], ...infer Remain extends readonly unknown[]]
+			? 0 extends Remain['length']
+				? SameLengthPositiveNumericStringGt<A, B>
+				: true
+			: false
+		: never;
+
+/**
+Returns a boolean for whether A > B(A and B are both positive numeric char).
```

Comment:
The descriptions for these types still needs some work.

Reply:
updated